search/search.js

```diff
@@ -0,0 +1,38 @@
+function div(a, b) {
+  return ((a + b)-(a + b)%2)/2;
+}
+
+function binarySearch(array, a) {
+  var mid_prev = -1;
+  var left = 0;
+  var right = array.length;
+  var mid = div(left, right);
+  while (true) {
```

Comment:
it's a bit strange to use `while (true)`, you can create a condition here, it's 100500 possible